openstudiocore/src/model/MeterCustom.cpp

```diff
@@ -0,0 +1,298 @@
+      return !insertExtensibleGroup(index, values).empty();
+    }
+    OS_ASSERT(false);
+    return false;
+  }
+
+  /** Get the Output Variable of Meter Name at index. Indexing starts at 0. */
+  boost::optional<std::string> MeterCustom_Impl::outputVariableorMeterName(unsigned index) const {
+    IdfExtensibleGroup eg = getExtensibleGroup(index);
+    if (!eg.empty()) {
+      return eg.getString(OS_Meter_CustomExtensibleFields::OutputVariableorMeterName, true);
+    }
+    return boost::none;
+  }
+
+  /** Set the Output Variable of Meter Name at index. Indexing starts at 0. */
+  bool MeterCustom_Impl::setOutputVariableorMeterName(unsigned index, const std::string& str) {
+    IdfExtensibleGroup eg = getExtensibleGroup(index);
+    if (!eg.empty()) {
+      return eg.setString(OS_Meter_CustomExtensibleFields::OutputVariableorMeterName, str);
+    } else {
+      StringVector values(2u);
+      values[OS_Meter_CustomExtensibleFields::OutputVariableorMeterName] = str;
+      return !insertExtensibleGroup(index, values).empty();
+    }
+    OS_ASSERT(false);
+    return false;
+  }
+
+
+
+
+} // detail
+
+MeterCustom::MeterCustom(const Model& model)
+  : ModelObject(MeterCustom::iddObjectType(), model)
+{
+  OS_ASSERT(getImpl<detail::MeterCustom_Impl>());
+
+  // Default the fuelType to Electricity (maybe "Generic"?)
+  setFuelType("Electricity");
+
+}
+
+IddObjectType MeterCustom::iddObjectType() {
+  return IddObjectType(IddObjectType::OS_Meter_Custom);
+}
+
+std::vector<std::string> MeterCustom::fuelTypeValues() {
+  return getIddKeyNames(IddFactory::instance().getObject(iddObjectType()).get(),
+                        OS_Meter_CustomFields::FuelType);
+}
+
+boost::optional<std::string> MeterCustom::fuelType() const {
+  return getImpl<detail::MeterCustom_Impl>()->fuelType();
+}
+
+bool MeterCustom::setFuelType(std::string fuelType) {
+  return getImpl<detail::MeterCustom_Impl>()->setFuelType(fuelType);
+}
+
+void MeterCustom::resetFuelType() {
+  getImpl<detail::MeterCustom_Impl>()->resetFuelType();
+}
+
+// Add a new (Key, Var) group
+bool MeterCustom::addKeyVarGroup(const std::string& keyName, const std::string& outputVariableorMeterName) {
+  return getImpl<detail::MeterCustom_Impl>()->addKeyVarGroup(keyName, outputVariableorMeterName);
+}
+
+// Remove the (Key, Var) group at given index
+bool MeterCustom::removeKeyVarGroup(unsigned groupIndex) {
+  return getImpl<detail::MeterCustom_Impl>()->removeKeyVarGroup(groupIndex);
+}
+
+// Remove all the (Key, Var) groups
+void MeterCustom::removeAllKeyVarGroups() {
+  getImpl<detail::MeterCustom_Impl>()->removeAllKeyVarGroups();
+}
+
+// // Return a vector of (Key, Var) pairs
+std::vector< std::pair<std::string, std::string> > MeterCustom::keyVarGroups() {
+  return getImpl<detail::MeterCustom_Impl>()->keyVarGroups();
+}
+
+// Return the number of (KeyName, OutputVariableorMeterName) groups
+unsigned MeterCustom::numKeyVarGroups() const {
```

Comment:
@jmarrec sometimes calling it `variable` other times `var`, can we make this consistent?